### PR TITLE
cleanupTimer should be disposed, making OutboxTask disposable

### PR DESCRIPTION
**Pull request by synhershko**
Following up on comments in https://github.com/Particular/NServiceBus.RavenDB/pull/43, porting relevant changes to NSB.NHibernate as well

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
Is there a case whre `Dispose` is called without prior calling `OnStop`? If so, why in `Dispose` we don't wait for the job to finish as in `OnStop`?

File: src/NServiceBus.NHibernate/Outbox/NHibernateOutboxStorage.cs [hunk 1]
```diff
@@ -92,10 +94,14 @@ void PerformCleanup(object state)
             {
                 OutboxPersister.RemoveEntriesOlderThan(DateTime.UtcNow - timeToKeepDeduplicationData);
             }
+
+            public void Dispose()
```
Comment (SzymonPobiega) [inline src/NServiceBus.NHibernate/Outbox/NHibernateOutboxStorage.cs]: Who not via Janitor?

**Comment by synhershko** [comment]
@SzymonPobiega see discussion at https://github.com/Particular/NServiceBus.RavenDB/pull/43/files#r24774858

I think the intent is to leverage the containers used, @SimonCropp is the guy to ask why not to rely on Janitor since it was his idea

**Comment by johnsimons** [comment]
What is the status of this one ?

File: src/NServiceBus.NHibernate/Outbox/NHibernateOutboxStorage.cs (hunk 1 above)
Comment (andreasohlund) [inline src/NServiceBus.NHibernate/Outbox/NHibernateOutboxStorage.cs]: @SzymonPobiega is this still relevant? close ? pull?

**Comment by andreasohlund** [comment]
@SzymonPobiega do we need this? should we close?

**Comment by SzymonPobiega** [comment]
@andreasohlund no idea. Let's close and re-open if applicable?

**Pull request closed without merging**